### DISCUSSION TITLE
Any way to setup sunshine to run on a system level, not the user level?

### DISCUSSION THREAD

**Discussion by hradec** [LizardByte/community#13] [Sunshine] [answered]
before anything, huge thanks for working on this!! I can't believe we finally have a decent low latency remote desktop on linux!!! Kudos!!!

So, I'm wondering if it's possible to configure sunshine to launch independently of the logged use in linux. Ideally, I would like to have sunshine serving the GDM screen, and later serving the desktop for anyone who logs into gdm! 

Is that setup possible in any way? 

It would also be good to remove the PIN setup, since we want to just give access to the machines in a LAN network to people connecting to a zerotier vpn... so there's no need to setup individual pin's for everyone. (specially because we want people to login using gdm over sunshine!)

make sense?

**Comment by ReenigneArcher** [comment, marked as answer]
The pin cannot removed without custom builds of moonlight client apps. Sunshine replicates GeForce Experience, so I don't see the pin method going away.

Regarding running as a different user... Have you tried `su` (substitute user) or `sudo`?

```bash
su - root -c "./sunshine"
OR
sudo ./sunshine
```

❗ Running as root/superuser/etc. is not advised.

**Reply by hradec** [reply]
> The pin cannot removed without custom builds of moonlight client apps. Sunshine replicates GeForce Experience, so I don't see the pin method going away.

Too bad. Although having a custom  moonlight client could be an interesting option! Thanks! 

> Regarding running as a different user... Have you tried su (substitute user) or sudo?

Indeed! But there's some permission issues when doing that. It seems sunshine can't access the X display when running as root since the X belongs to another user. Even by setting DISPLAY to the correct X number won't work. I guess the only option would be to setup sunshine for the GDM user (so it would run under GDM user when no-one is logged in and GDM is showing the login screen), but then sunshine would have to be restarted for the current logged user, when someone logs in... and in such scenario, the pin "thing" becomes a nightmare.

I'm running sunshine in fedora35 right now, and it's not finding NVFBC aparently. Maybe with NVFBC the X permission wouldn't be an issue? (since it would be grabing the framebuffer straight from the GPU, it shouldn't care about what procces is "drawing" on it maybe?)
 
So, I'll try to make sunshine work correctly with NVFBC and give it another try as root. 

> exclamation Running as root/superuser/etc. is not advised.

I understand. Although as we're planning on run it in an isolated LAN (no internet access even), only accessible through a VPN bridge, it shouldn't be a problem.

thanks lots for the fast reply! Really appreciate!

**Comment by qdrop17** [comment]
I would like to setup Sunshine in a way that it's starting prior to my GDM3 login manager (Fedora 38 with Gnome on Wayland). That way, I can connect to my host and login to to my session with low latency and a high refresh rate.

Sunshine somehow has to capture the output of my GPU (NVIDIA). But for this output some display manager has to be started right?
How can this be done? I did not find any resources to do so.

I would also be fine with some command to start Sunshine on demand (including a Wayland session). But I wasn't able to find resources on how to accomplish that.
